etcdserver: forbid to unset v3 demo once used

yichengq opened the pull request:
After enabling v3 demo, it may change the underlying data organization
for v3 store. So we forbid to unset --experimental-v3demo once it has
been used.

xiang90 commented on etcdserver/server.go:
```diff
@@ -180,6 +181,10 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 	var cl *cluster
 
+	if !cfg.V3demo && isFileExist(cfg.StorageDir()) {
+		return nil, errors.New("v3 should always be enabled if it has run once before")
```
experimental-v3demo cannot be disabled once it was enabled.

yichengq commented:
Updated sentence and fixed inconsistent code.

When I read it again, I find that there is no need to change the file to directory in this PR, so I remove this change.

xiang90 commented on etcdserver/config.go:
```diff
@@ -119,6 +119,8 @@ func (c *ServerConfig) WALDir() string {
 
 func (c *ServerConfig) SnapDir() string { return path.Join(c.MemberDir(), "snap") }
 
+func (c *ServerConfig) V3DemoFile() string { return path.Join(c.MemberDir(), "v3demo") }
```
Can we remove this? I do not want to attach more temp stuff on the config.

yichengq replied:
ok

xiang90 commented on etcdserver/storage.go:
```diff
@@ -141,3 +141,8 @@ func makeMemberDir(dir string) error {
 	return nil
 }
+
+func isExist(name string) bool {
```
move this to fileutil please.

yichengq commented:
updated. And I add some test around new function in fileutil pkg.

xiang90 commented on pkg/fileutil/fileutil.go:
```diff
@@ -55,3 +55,8 @@ func ReadDir(dirpath string) ([]string, error) {
 	return names, nil
 }
+
+func IsExist(name string) bool {
```
Exist?

yichengq replied:
Exist looks shorter and expresses the functionality. ok.

yichengq commented:
Updated. It looks good now.

xiang90 commented:
LGTM. When we want to save the snapshot into this, we have to move the db file into a formal directory.
